File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_hashmap = {}
        for i, x in enumerate(nums):
            if x in index_hashmap:
                return [index_hashmap[x], i]
            complement = target - x
            index_hashmap[complement] = i

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer,Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
                int test = target - nums[i];
                if (map.containsKey(test)){
                    result[1] = i;
                    result[0] = map.get(test);
                    return result;
                }
                map.put(nums[i], i);
            }
        return null;
    }
}